widgets/clear-all/clear-all.js

```diff
@@ -0,0 +1,95 @@
+let React = require('react');
+let ReactDOM = require('react-dom');
+
+let {bemHelper, getContainerNode, prepareTemplateProps} = require('../../lib/utils.js');
+let bem = bemHelper('ais-clear-all');
+let cx = require('classnames');
+
+let autoHideContainer = require('../../decorators/autoHideContainer');
+let headerFooter = require('../../decorators/headerFooter');
+
+let isEmpty = require('lodash/lang/isEmpty');
+
+let defaultTemplates = require('./defaultTemplates');
+
+/**
+ * Allows to clear all refinements at once
+ * @param  {string|DOMElement} options.container CSS Selector or DOMElement to insert the widget
+ * @param  {Object} [options.cssClasses] CSS classes to add
+ * @param  {string|string[]} [options.cssClasses.root] CSS class to add to the root element
+ * @param  {string|string[]} [options.cssClasses.header] CSS class to add to the header element
+ * @param  {string|string[]} [options.cssClasses.body] CSS class to add to the body element
+ * @param  {string|string[]} [options.cssClasses.footer] CSS class to add to the footer element
+ * @param  {string|string[]} [options.cssClasses.link] CSS class to add to the link element
+ * @param  {Object} [options.templates] Templates to use for the widget
+ * @param  {string|Function} [options.templates.header=''] Header template
+ * @param  {string|Function} [options.templates.link] Link template
+ * @param  {string|Function} [options.templates.footer=''] Footer template
+ * @param  {boolean} [options.autoHideContainer=true] Hide the container when there's no refinement to clear
+ * @return {Object}
+ */
+function clearAll({
+    container,
+    templates = defaultTemplates,
+    cssClasses: userCssClasses = {},
+    autoHideContainer: doAutoHideContainer = true
+  } = {}) {
+  let containerNode = getContainerNode(container);
+  let usage = 'Usage: toggle({container[, cssClasses.{root,header,body,footer,link}, templates.{header,link,footer}, autoHideContainer]})';
+
+  let ClearAll = headerFooter(require('../../components/ClearAll/ClearAll.js'));
```

Comment:
I'd say yes because it can definetely be included in the side bar with other facets widgets and you might want to style them all the same way.